pkg/qrcode/lora_alliance_tr005.go

```diff
@@ -0,0 +1,101 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package qrcode
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"go.thethings.network/lorawan-stack/pkg/types"
+)
+
+// LoRaAllianceTR005Draft2 is the LoRa Alliance defined format in Technical Recommendation TR005 Draft 2.
+type LoRaAllianceTR005Draft2 struct {
+	JoinEUI,
+	DevEUI types.EUI64
+	VendorID             [2]byte
+	ModelID              [2]byte
+	DeviceValidationCode []byte
+	SerialNumber,
+	Proprietary string
+}
+
+// MarshalText implements the TextMarshaler interface.
+func (m LoRaAllianceTR005Draft2) MarshalText() ([]byte, error) {
+	var ext string
```

Comment:
Could consider using a `strings.Builder` here

Reply:
Doesn't get more elegant and we're probably close to I/O or rendering here anyway